components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -1899,4 +1901,22 @@ private static Class<? extends OAuthAuthzRequest> getOAuthAuthzRequestClass() th
         }
         return oAuthAuthzRequestClass;
     }
+
+    /**
+     * Validate the response mode against the response type as per FAPI spec.
+     * Response mode required to be jwt if code response type is used.
+     * <a href="https://openid.net/specs/openid-financial-api-part-2-1_0.html#authorization-server">5.2.2-2.2</a>
+     *
+     * @param responseType response mode
+     * @param responseMode response type
+     * @throws OAuthProblemException when response mode is not valid
+     */
+    public static void validateFAPIResponseMode(String responseType, String responseMode)
```

Comment:
lets change the method name in the util to give more clarity. Current method name reads like that there is a response mode named "FAPI"

```suggestion
    public static void validateFAPIAllowedResponseMode(String responseType, String responseMode)
```

Reply:
Also doesn't the spec say to validate having `code id_token` in https://openid.net/specs/openid-financial-api-part-2-1_0.html#authorization-server `2.1`?

Reply:
Also there should be a unit test for this.

Reply:
Fixed in 9ed19436be3c36b2fe703d66e088386ce1c78c86